core/common/src/IModel.ts

```diff
@@ -184,10 +187,22 @@ export class EcefLocation implements EcefLocationProps {
   public readonly orientation: YawPitchRollAngles;
   /** Optional position on the earth used to establish the ECEF origin and orientation. */
   public readonly cartographicOrigin?: Cartographic;
+  /** Optional, column vectors for ECEF non-rigid ECEF transform  */
+  public readonly xVector?: Vector3d;
+  public readonly yVector?: Vector3d;
 
   /** Get the transform from iModel Spatial coordinates to ECEF from this EcefLocation */
   public getTransform(): Transform {
-    return Transform.createOriginAndMatrix(this.origin, this.orientation.toMatrix3d());
+    let matrix;

```

Comment:
This gets called quite frequently from the frontend, right? Did you consider computing the transform once in the constructor?

Reply:
Done.